### ISSUE TITLE
[task] port to minetest 5.x engines

### ISSUE THREAD

**Issue by programmerjake** [D00Med/LegendofMinetest#40]
It would be nice to be able to play this on modern versions of minetest

**Comment by D00Med** [comment]
I no longer work on this project (or any of my github projects). 
But as nice as it would be to have a new version, it might be better for this particular project not to be revived, since it's a fangame.

**Comment by kveroneau** [comment]
While yes, it's technically a fangame, it is not using any actual assets from the actual games, nor is it trying to recreate any of the games.  I feel this mod falls under "Fair Use".  I'm honestly still surprised that Zelda Classic is still kicking, as they remade the original game, and are using assets from the original game as well.  So, not all fangames get hit by the big N.  I recently recreated the original Zelda as a text-adventure game. :D https://github.com/kveroneau/InteractiveFiction/tree/main/Hyrule  Playable here:  http://tech406.com/hyrule.xml

**Comment by D00Med** [comment]
Even so, it will never be able to be promoted in the same way other games are (eg: linked from the minetest website).

**Comment by mckaygerhard** [comment]
hi @D00Med  what happened with this #39  ? you are owner now! i have plans to do some update fixeds before porting to 5.X (i mean still made playable in 0.4 before fixed for 5.X) but i need the pull request to be merged!

i also port your dmobs fixed  and up to dated with some commits and changed namespaced to mobs_doomed.. please write here if you can checks pending merges to start to work with this repo to make it work again

**Comment by D00Med** [comment]
I don't know...I guess this was the last thing I worked on before I gave up working on minetest things. 
I will merge the pull request. After that, you could fork the project so you can have full control.

**Comment by mckaygerhard** [comment]
i will only provide easy to review pull request to you.. i want to avoid forking control due is duplication of work.. its better to continue the work so others can help too, thanks @D00Med

**Comment by programmerjake** [comment]
@D00Med if you want to have someone officially take over this project, you can transfer the repo to their account, or to a github organization, which will make github redirect to the new repo location anytime someone uses a link to the old repo or accesses the repo using git commands, as well as keeping all the issues and pull requests and other history.

https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository

**Comment by D00Med** [comment]
-_- Sounds like a lot of effort and work for something I stopped working on 4 years ago. 

I can add people as contributors so they can merge PR's on their own (or blindly merge them myself). But if someone makes a fork, my original work is preserved and available for others to fork off if they don't like the newer forks.

**Comment by D00Med** [comment]
Pretty sure it is Minetest tradition to palm everyone off with "make your own fork"

**Comment by mckaygerhard** [comment]
> @D00Med if you want to have someone officially take over this project, you can transfer the repo to their account, or to a github organization, which will make github redirect to the new repo location anytime someone uses a link to the old repo or accesses the repo using git commands, as well as keeping all the issues and pull requests and other history.
> 
> https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository

your idea sound good but we need two reviewers, currently there'0s only one person in interest.. so i proposed this as @D00Med  said:

> I can add people as contributors so they can merge PR's on their own. But if someone makes a fork, my original work is preserved and available for others to fork off if they don't like the newer forks.

the idea of PR contributions, and i always preserved the original main master branch of original work.. as the minenux repos demostrates,  in your idea, i will mantain a main master focused on estability until completely works stable (currently i started in my fork)

UPDATED: This conservational work will end when the game repo works perfect in MT0.4 (my current interest and working one) and then i start to port to MT 5.X, at that moment others will be intereste and another contributor will be born.. i will provide documentation and issue attention only in this moment of course..

**Comment by D00Med** [comment]
You want to be added as a contributor then?

**Comment by mckaygerhard** [comment]
hi @D00Med  yes.. with possibility to manage the issue but i ask :

* there's a possibilty to you that always review all the pull request i made it, or any other so? i know maybe you dont have anymore the game in situ but at leas to review the description and code in general ..

if there's no possiblity to, then i need PR permissions and issues organization permissions with:

* my deal is first to be playable in MT4 (currently i only will upgrade some mobs to work with las 0.4.17 due 99% of the phones uses the 0.ñ4 minetst api, sorry is a crude reality), also for 0.4.X there's many packages of 0.4.X for most linux distros, and i have an updated 0.4.X branch, and also 5.2.X either
* i will take almost 3 phases, for phase 1 and 2 if in such time some issue is for 5.X or update, i will try to merge but with 0.4.X compatibilty work too.. 
* phase 1 is knowledge and try to up to date the current game (so that's why 0.4.X) by example i have a question.. how can i put the armor, i give me one in creative but it seems i cannot use it! @D00Med .. of course in each phase we always try to fix any bug
* phase 2 is focused on bugfixed and compatibilty of both engiines, also try to fix game focused things.. like the bug of Shelves or booksellers (that cannot interact if are not put by player)
* phare 3, after that time passed the 5.X compatibilty wil be the priority and 0.4.X will be branched and deprecated, but still i started to port to 5.X right now as i said in the issue about the player update model, player api was changed but i know how to fix that
* i only focused in linux playability or phones.. any other OS will be reviewed by you or other people, and then later i will merge, but i never merge a PR for other OS that is not reviewed by more than one.. due we two are the only ones.. i can wait for those very active .. like wuzzy or smalljoker for .. and after he writes a comment i will merge, OR you can reviewed and merge that!

I think is not so bad those demands.. in fact are very soft and there's nothig to lost..

**Comment by D00Med** [comment]
That is fine, I only ask that you don't change textures or names (apart from correcting typos). 

There's a few things you've mentioned now which are not really 'bugs' but are actually intended behaviour. 
For example, the armour cannot be worn in creative, because the player doesn't need it. If you disable creative, you can wear armour. This is how the original armour mod works. 
Similarly, the items without descriptions have no descriptions because they are not supposed to be accessible to the player. Many mods do this for items that work behind the scenes or as duplicates (for example: 3d models for tools, or the tower spawning blocks). I think most of the tower blocks are like this (and also unbreakable to force the player to find keys instead of digging).

Somewhere in hyruletools, I think there is a command or node that gives you the developer tools like a pick that destroys anything or a few of the blocks not accessible through the creative menu.

**Comment by mckaygerhard** [comment]
i will then focused in this repo for the next 2 years.. and inclusivelly try to release a public server for.. 

- [X] i never change textures.. neither names.. so that's why #42 i reconiced that you said.. and posted that is not priority.. so i will close that!
- [X]  3d armor marked as new task due _canot be worm in creative_ so when we ported to 5.X we need to check behaviour.. so moved to #43 
- [X] items behaviour documentation for better novice understand.. i was a average gamer of the legen of zelda.. was not played so deep but i will ask to my brothers that still handle original games, moved documentation of repository to #44

> Somewhere in hyruletools, I think there is a command or node that gives you the developer tools like a pick that destroys anything or a few of the blocks not accessible through the creative menu.

i already check the code and also said its not finished or xperimental or something.. i can guess its an old comment .. so this part will depends of #44 to get more contributors.. 

> I think most of the tower blocks are like this (and also unbreakable to force the player to find keys instead of digging).

already noted and that will depedns on #44 due players must understand two things:

* this game is a inspired in zelda that is RPG so player in main target must "do" something, not complety free.-. 
* this game due the previous.. have special settings so new contributors will need to understand the changes

also i noted that i must study more in deep the game api.. puffff

**about this issue: lest mantain stalled with no comments until i got more or try to port most of the mods.. i will try to get this in soe months..**  every new issue related to porting to 5.X i will try to get linked here!

**Comment by D00Med** [comment]
Thankyou so much for your interest in this. 
What you've said makes sense and makes a good roadmap.

**Comment by mckaygerhard** [comment]
Hi @D00Med  i was pretty busy.. also foudn that you dedicated so many time and skills to the game.. the 3darmor.. the specific mods made for the game and the modified derived ones are so hard to follow withoyut a good OBJ OOP and knowledge of minetest.. 

now i know how to made a mod and found lof of bugs already .. also .. noted the usage of new and older apis.. work with some developers of some mods..and i guess this will be more tedous and large thatn i think.. so if you feels i am so "quiet" or to be so much silence here.. its not that you think.. i am pretty ready and take attention to this project:

**meanwhile I will only limit myself to ordering and documenting the code since changing/featured it requires more experience with the game but even if I do not add new things I will try to make them work in v5 .. I hope you understand .. I repeat I am not forgetting this .. I just 'll go slower**

**Comment by mckaygerhard** [comment]
Hi @D00Med i was pretty busy.. also foudn that you dedicated so many time and skills to the game.. the 3darmor.. the specific mods made for the game and the modified derived ones are so hard to follow withoyut a good OBJ OOP and knowledge of minetest.. 

now i know how to made a mod and found lof of bugs already .. also .. noted the usage of new and older apis.. work with some developers of some mods..and i guess this will be more tedous and large thatn i think.. so if you feels i am so "quiet" or to be so much silence here.. its not that you think.. 

i am pretty ready, try to be active and take attention to this project: all the work done around the mods was put in codeberg.. at https://codeberg.org/minenux but as i said.. you changed lof of thing.. so i need time to assimilate the things ...

**meanwhile I will only limit myself to ordering and documenting the code since changing/featured it requires more experience with the game but even if I do not add new things I will try to make them work in v5 .. I hope you understand .. I repeat I am not forgetting this .. I just 'll go slower**

**Comment by mckaygerhard** [comment]
@programmerjake  after a time i was able to port some mods and there's 3 version to care.. :

1. from 5.0 there's a player api slipted from default mod
2. from 5.3 the creative check is from a mod with a engine check api
3. from 5.4 there's a new way to access some new callbacks and api calls

i has all the new changes to 5.0 to 5.3 releases but for callbacks chages to 5.4 will demand more time.. i will port in following months the usage on 5.0 5.1 and 5.2.. sorry for long wait.. 

Related tasks:

* #45 
* #47 
* #49
